modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php

```diff
@@ -110,6 +126,63 @@ protected function runIt() {
     return $this->getResult();
   }
 
+  /**
+   * Read the first line from the given file.
+   *
+   * @param string $file_path
+   *   File path.
+   *
+   * @return string
+   *   First line from file.
+   *
+   * @throws Symfony\Component\HttpFoundation\File\Exception\FileException
+   *   On failure to open the file;
+   *   on failure to read the first line from the file.
+   */
+  protected function getFirstLineFromFile(string $file_path): string {
+    // Read the first (header) line from the CSV file.
+    $f = fopen($file_path, 'r');
+    // Ensure the file could be successfully opened.
+    if (!isset($f) || $f === FALSE) {
+      throw new FileException(sprintf('Failed to open resource file "%s".', $file_path));
+    }
+    // Attempt to retrieve the first line from the resource file.
+    $header_line = fgets($f);
```

Comment:
Note that auto-detect-line-endings is disabled by default - not site how typical it is to configure it?
https://www.php.net/manual/en/filesystem.configuration.php#ini.auto-detect-line-endings

Reply:
@grugnog Good point! I've updated the the `MysqlImport::getFirstLineFromFile()` method to temporarily enable this ini setting before opening the file.

Reply:
Looks good - does this setting persist through the fgets() that actually reads what it deems a "line"... perhaps you tested, but the docs aren't clear!

Reply:
Yes, I suppose this is undocumented behavior which could potentially change, but the newline detection behavior appears to currently be determined during `fopen()`.